server/src/main/java/org/elasticsearch/index/mapper/vectors/VectorDVLeafFieldData.java

```diff
@@ -76,4 +87,109 @@ public DocValuesScriptFieldFactory getScriptFieldFactory(String name) {
+                    return values.advanceExact(docId);
+                }
+
+                @Override
+                public int docValueCount() throws IOException {
+                    return values.docValueCount();
+                }
+
+                @Override
+                public Object nextValue() throws IOException {
+                    return values.nextValue();
+                }
+            };
+            case FLOAT -> new FormattedDocValues() {
+                final DenseVectorNumericFloatValues values = getFloatValues();
+
+                @Override
+                public boolean advanceExact(int docId) throws IOException {
+                    return values.advanceExact(docId);
+                }
+
+                @Override
+                public int docValueCount() throws IOException {
+                    return values.docValueCount();
+                }
+
+                @Override
+                public Object nextValue() throws IOException {
+                    return values.nextValue();
+                }
+            };
+        };
+    }
+
+    private DenseVectorNumericByteValues getByteValues() {
+        int dims = elementType == ElementType.BIT ? this.dims / Byte.SIZE : this.dims;
+        return new DenseVectorNumericByteValues(dims) {
+            @Override
+            public boolean advanceExact(int docID) throws IOException {
+                if (indexed) {
+                    final ByteVectorValues byteVectorValues = reader.getByteVectorValues(field);
+                    if (byteVectorValues.advance(docID) == NO_MORE_DOCS) {
+                        return false;
+                    }
+                    values = byteVectorValues.vectorValue();
+                } else {
+                    BinaryDocValues binary = DocValues.getBinary(reader, field);
+                    if (binary.advance(docID) == NO_MORE_DOCS) {
+                        return false;
+                    }
+                    BytesRef ref = binary.binaryValue();
+                    ByteBuffer byteBuffer = ByteBuffer.wrap(ref.bytes, ref.offset, ref.length);
+                    if (indexVersion.onOrAfter(DenseVectorFieldMapper.LITTLE_ENDIAN_FLOAT_STORED_INDEX_VERSION)) {
+                        byteBuffer.order(ByteOrder.LITTLE_ENDIAN);
+                    }
+                    values = new byte[dims];
+                    for (int dim = 0; dim < dims; dim++) {
+                        values[dim] = byteBuffer.get();
+                    }
+                }
+                valuesCursor = 0;
+                return true;
+            }
+        };
+    }
+
+    private DenseVectorNumericFloatValues getFloatValues() {
+        return new DenseVectorNumericFloatValues(dims) {
+            @Override
+            public boolean advanceExact(int docID) throws IOException {
+                if (indexed) {
+                    final FloatVectorValues floatVectorValues = reader.getFloatVectorValues(field);
```

Comment:
You should not instantiate vector values on every iteration of docID.
They should be instantiated only once:

try {
            FloatVectorValues vectorValues = reader.getFloatVectorValues(field);

        } catch (IOException e) {
            throw new IllegalStateException("Cannot load doc values", e);
        }
and then use the save instance for iteration.

Reply:
Thanks, @mayya-sharipova! Oops, that was a rookie mistake. Fixed it now—only instantiating once and reusing. Appreciate the catch!